ionoscloud/resource_server.go

```diff
@@ -568,65 +568,62 @@ func resourceServerCreate(ctx context.Context, d *schema.ResourceData, meta inte
 
 	serverType := d.Get("type").(string)
 
+	serverReq.Entities = ionoscloud.NewServerEntities()
+
 	// create volume object with data to be used for image
 	volume, err := getVolumeData(d, "volume.0.", serverType)
-
-	if err != nil {
-		return diag.FromErr(err)
-	}
-
-	// get image and imageAlias
-	image, imageAlias, err := getImage(ctx, client, d, *volume)
 	if err != nil {
 		return diag.FromErr(err)
 	}
+	if volume.Type != nil && *volume.Type != "" {
```

Comment:
@cristiGuranIonos can you also please add a test in which we create a server without a volume? Also maybe, if it's not to complex, we can add some simple edge cases.